autobuild/workflow.md

```diff
@@ -0,0 +1,195 @@
+# Coin update workflow
+
+This document attempts to describe the process for updating or deleting an existing coin configuration or adding a new coin. The information required has not changed from the old process but the mechanism by which it is consumed has been changed. In an effort to increase efficiency and reduce the liklihood of mistakes we have automated as much of the process as possible.
+
+Adding or updating a coin definition in the blockchain-configuration-files repository is only the first part of the process. The configuration files created here are consumed by different processes, including
+
+* Docker image creation automation
+* EXR proxy deployment automation
+* BlockDX configuration
+
+The entire "automation" workflow is spread over several repositories. This documentation is a preliminary WIP and feedback is welcome to help improve it.
+
+## Automation functions
+
+The sources of truth for BlockDX coin configurations are the following files within this repository: 
+| Path | purpose |
+|------|---------|
+| manifest-latest.json | master ultimate authority |
+| autobuild/configs/*.base.j2 | ultimate authority |
+| autobuild/templates/*j2 | auxiliary templates |
+| wallet-confs/*.conf | public facing config fragments |
+| xbridge-confs/*.conf | public facing config fragments |
+
+Ultimate authority files are created manually or by automation workflows while public facing config fragments are generated by automation workflows. In the past the public facing config fragments were manually generated which allowed mistakes to creep in. The automation workflows in this and other repositories attempt to speed up consumption processes and reduce errors.
+
+## Ultimate authorities
+The "sources of truth" upon which BlockDX depends are the
+* manifest-latest.json
+* autobuild/configs/*.base.j2 
+
+files. Ultimately automation may be available to generate these too given only a coin name and Github URL, but for now they must be manually created. The old instructions for adding/updating a coin provide the necessary information but the locations for some of it have changed. 
+
+These sources of truth are manipulated/consumed by four automation scripts in the autobuild directory of this blockchain-configuration-files repository.
+
+## Automation scripts
+
+There is one primary script intended for "daily" use and three auxilliary scripts located in the utils sub-directory intended for "special" use tasks. 
+### app.py
+**Purpose:**
+builds individual coin wallet.conf and xbridge.conf fragments from the new format .j2 master sources-of-truth
+
+**Pre-reqs:**
+* manifest-latest.json
+* autobuild/configs/*.base.j2
+* autobuild/templates/*.j2
+
+**Execution:**
+within your local repo/autobuild directory, for example:
+```
+mark@x230:~/src/blockdx-configs$ cd autobuild
+mark@x230:~/src/blockdx-configs/autobuild$ python3 app.py
+```
+
```

Comment:
It would probably be good to mention here that `app.py` with no params will build `wallet.conf` and `xbridge.conf` for *all* coins in `manifest-latest.json`, and that the user can pass an optional param `--coins COINS`  (COINS = List of coins. Eg.: BTC, LTC, DASH) to specify a limited set of coins for which `wallet.conf` and `xbridge.conf` files should be generated.

Reply:
Done